market_maker_keeper/gas.py

```diff
@@ -15,22 +15,61 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from argparse import Namespace, ArgumentParser
 from typing import Optional
 
-from ethgasstation_client import EthGasStation
-from pymaker.gas import GasPrice, IncreasingGasPrice, FixedGasPrice, DefaultGasPrice
+from pymaker.gas import GasPrice, GeometricGasPrice, IncreasingGasPrice, FixedGasPrice, DefaultGasPrice, NodeAwareGasPrice
+from pygasprice_client import EtherchainOrg, EthGasStation, POANetwork
+from pygasprice_client.aggregator import Aggregator
+from web3 import Web3
+
+def add_gas_arguments(parser: ArgumentParser):
+    gas_group = parser.add_mutually_exclusive_group()
+
+    gas_group.add_argument("--smart-gas-price", dest='smart_gas_price', action='store_true',
+                        help="DEPRACATED use dynamic-gas-price instead where possible. Use smart gas pricing strategy, based on the ethgasstation.info feed.")
+
+    gas_group.add_argument("--dynamic-gas-price", dest='dynamic_gas_price', action='store_true',
+                        help="Use dynamic gas pricing strategy, based on pygasprice-client.aggregator")
+
+    parser.add_argument("--oracle-gas-price", action='store_true',
+                            help="Use a fast gas price aggregated across multiple oracles")
+
+    parser.add_argument('--fixed-gas-price', type=float, default=20,
+                           help="Uses a fixed value (in Gwei) instead of an external API to determine initial gas")
+
+    parser.add_argument("--ethgasstation-api-key", type=str, default=None, help="ethgasstation API key")
+
+    parser.add_argument("--etherscan-api-key", type=str, default=None, help="etherscan API key")
+
+    parser.add_argument('--etherchain-gas-price', dest='etherchain_gas', action='store_true',
+                           help="Use etherchain.org gas price")
+
+    parser.add_argument('--poanetwork-gas-price', dest='poanetwork_gas', action='store_true',
+                           help="Use POANetwork gas price")
+
+    parser.add_argument("--poanetwork-url", type=str, default=None, help="Alternative POANetwork URL")
+
```

Comment:
The `gasnow` oracle is also supported.  Was likely implemented after you started work on these changes.  The `oracle-gas-price` captures it.

Reply:
Not sure i'm following - is there an additional argument I should be adding?

Reply:
Perhaps; I didn't see a `--gasnow` argument and wondered if it was intentionally omitted.